Objective-C/CBLDatabase.mm

```diff
@@ -360,58 +360,83 @@ - (BOOL) inBatch: (NSError**)outError usingBlock: (void (NS_NOESCAPE ^)())block
 #pragma mark - DATABASE MAINTENANCE
 
 - (BOOL) close: (NSError**)outError {
+    NSArray *activeReplicators, *activeLiveQueries = nil;
+    
     CBL_LOCK(self) {
-        if (_c4db == nullptr)
+        if ([self isClosed] || _isClosing)
```

Comment:
This seems a bit dangerous in that two rapid calls will result in the second one returning `YES` before the database is actually closed.  I think any subsequent calls should block until the close condition, and if possible report back the ultimate success of the operation from the other thread.

Reply:
Really good point. I will fix this and update the technical spec regarding this well.